lmdeploy/pytorch_poc/patch/baichuan.py

```diff
@@ -0,0 +1,327 @@
+                position_ids=position_ids,
+                past_key_value=past_key_value,
+                output_attentions=output_attentions,
+                use_cache=use_cache,
+            )
+
+            hidden_states = layer_outputs[0]
+
+            if use_cache:
+                next_decoder_cache += (
+                    layer_outputs[2 if output_attentions else 1], )
+
+            if output_attentions:
+                all_self_attns += (layer_outputs[1], )
+
+        hidden_states = origin_self.norm(hidden_states)
+
+        # add hidden states from the last decoder layer
+        if output_hidden_states:
+            all_hidden_states += (hidden_states, )
+
+        next_cache = next_decoder_cache if use_cache else None
+        if not return_dict:
+            return tuple(
+                v for v in
+                [hidden_states, next_cache, all_hidden_states, all_self_attns]
+                if v is not None)
+        return BaseModelOutputWithPast(
+            last_hidden_state=hidden_states,
+            past_key_values=next_cache,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attns,
+        )
+
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        past_key_values: Optional[List[torch.FloatTensor]] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutputWithPast]:
+        use_origin = self.context.use_origin
+
+        if input_ids is not None and inputs_embeds is not None:
+            raise ValueError('You cannot specify both decoder_input_ids '
+                             'and decoder_inputs_embeds at the same time')
+        elif input_ids is not None:
+            assert input_ids.dim() == 2
+        elif inputs_embeds is not None:
+            assert inputs_embeds.dim() == 3
+        else:
+            raise ValueError(
+                'You have to specify '
+                'either decoder_input_ids or decoder_inputs_embeds')
+
+        if use_origin:
+            # use origin model
+            return self.origin_mod(input_ids, attention_mask, position_ids,
+                                   past_key_values, inputs_embeds, use_cache,
+                                   output_attentions, output_hidden_states,
+                                   return_dict)
+        else:
+            return self._continuous_batching_forward(
+                input_ids, attention_mask, position_ids, past_key_values,
+                inputs_embeds, use_cache, output_attentions,
+                output_hidden_states, return_dict)
+
+
+class BaichuanForCausalLM(nn.Module):
+
+    def forward(self,
```

Comment:
Is this rewrite for https://github.com/baichuan-inc/Baichuan-7B/blob/6f3ef4633a90c2d8a3e0763d0dec1b8dc11588f5/models/modeling_baichuan.py#L581? I did not see much difference between this one and the origin one.

Reply:
It's for 13B